src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -657,12 +687,53 @@ impl<'index, D: Decoder> IndexReader<'index, D> {
         true
     }
 
+    /// Reset the reader to the beginning of the index.
+    pub fn reset(&mut self) {
+        self.set_current_block(0);
+    }
+
+    /// Return the number of unique documents in the underlying index.
+    pub fn unique_docs(&self) -> usize {
+        self.ii.unique_docs()
+    }
+
+    /// Returns true if the underlying index has duplicate document IDs.
+    pub fn has_duplicates(&self) -> bool {
+        self.ii.flags() & IndexFlags_Index_HasMultiValue > 0
```

Comment:
not a blocker but we should use `bitflags` at some point.

Reply:
Agree, it will make it easier to work with these